pkg/util/strategicpatch/patch.go

```diff
@@ -240,15 +602,20 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	return original, nil
 }
 
-// This panics if any element of the slice is not a map.
+// This method no longer panics if any element of the slice is not a map.
```

Comment:
I don't know what the impact of panics is.

I agree that if we don't panic, this needs to return an error.